src/inputs/TextFieldBase.tsx

```diff
@@ -60,15 +71,49 @@ export function TextFieldBase(props: TextFieldBaseProps) {
     onFocus,
     xss,
     endAdornment,
+    startAdornment,
     inlineLabel,
+    borderless = false,
+    contrast = false,
   } = props;
   const errorMessageId = `${inputProps.id}-error`;
   const labelSuffix = getLabelSuffix(required);
   const ElementType: React.ElementType = multiline ? "textarea" : "input";
-  const tid = useTestIds(props, defaultTestId(label || "textField"));
+  const tid = useTestIds(props, defaultTestId(label));
   const [isFocused, setIsFocused] = useState(false);
+  const { hoverProps, isHovered } = useHover({});
   const { focusWithinProps } = useFocusWithin({ onFocusWithinChange: setIsFocused });
 
+  const fieldHeight = 40;
+  const compactFieldHeight = 32;
+
+  const fieldStyles = {
+    container: Css.df.fdc.w100.maxw(px(550)).$,
+    inputWrapper: {
+      ...Css.sm.df.aic.br4.px1.w100.$,
+      ...Css.bgWhite.bGray300.gray900.if(contrast).bgGray700.bGray700.white.$,
+      // For borderless heights, remove the border from the element's height with the expectation the caller will apply a border if necessary (Example: DateFilter)
+      ...(borderless
+        ? Css.hPx(fieldHeight - 2)
+            .if(compact)
+            .hPx(compactFieldHeight - 2).$
+        : Css.ba.hPx(fieldHeight).if(compact).hPx(compactFieldHeight).$),
```

Comment:
Could maybe do `const maybeSmaller = borderless ? 2 : 0` and then `hPx(fieldHeight - maybeSmaller).if(compact(compactFieldHeight - maybeSmaller)`